mono/metadata/console-unix.c

```diff
@@ -358,66 +357,66 @@ console_restore_signal_handlers ()
 #endif
 
 static void
-set_control_chars (MonoArray *control_chars, const guchar *cc)
+set_control_chars (gchar *control_chars, const guchar *cc)
 {
 	/* The index into the array comes from corlib/System/ControlCharacters.cs */
 #ifdef VINTR
-	mono_array_set (control_chars, gchar, 0, cc [VINTR]);
+	control_chars [0] = cc [VINTR];
 #endif
 #ifdef VQUIT
-	mono_array_set (control_chars, gchar, 1, cc [VQUIT]);
+	control_chars [1] = cc [VQUIT];
 #endif
 #ifdef VERASE
-	mono_array_set (control_chars, gchar, 2, cc [VERASE]);
+	control_chars [2] = cc [VERASE];
 #endif
 #ifdef VKILL
-	mono_array_set (control_chars, gchar, 3, cc [VKILL]);
+	control_chars [3] = cc [VKILL];
 #endif
 #ifdef VEOF
-	mono_array_set (control_chars, gchar, 4, cc [VEOF]);
+	control_chars [4] = cc [VEOF];
 #endif
 #ifdef VTIME
-	mono_array_set (control_chars, gchar, 5, cc [VTIME]);
+	control_chars [5] = cc [VTIME];
 #endif
 #ifdef VMIN
-	mono_array_set (control_chars, gchar, 6, cc [VMIN]);
+	control_chars [6] = cc [VMIN];
 #endif
 #ifdef VSWTC
-	mono_array_set (control_chars, gchar, 7, cc [VSWTC]);
+	control_chars [7] = cc [VSWTC];
 #endif
 #ifdef VSTART
-	mono_array_set (control_chars, gchar, 8, cc [VSTART]);
+	control_chars [8] = cc [VSTART];
 #endif
 #ifdef VSTOP
-	mono_array_set (control_chars, gchar, 9, cc [VSTOP]);
+	control_chars [9] = cc [VSTOP];
 #endif
 #ifdef VSUSP
-	mono_array_set (control_chars, gchar, 10, cc [VSUSP]);
+	control_chars [10] = cc [VSUSP];
 #endif
 #ifdef VEOL
-	mono_array_set (control_chars, gchar, 11, cc [VEOL]);
+	control_chars [11] = cc [VEOL];
 #endif
 #ifdef VREPRINT
-	mono_array_set (control_chars, gchar, 12, cc [VREPRINT]);
+	control_chars [12] = cc [VREPRINT];
 #endif
 #ifdef VDISCARD
-	mono_array_set (control_chars, gchar, 13, cc [VDISCARD]);
+	control_chars [13] = cc [VDISCARD];
 #endif
 #ifdef VWERASE
-	mono_array_set (control_chars, gchar, 14, cc [VWERASE]);
+	control_chars [14] = cc [VWERASE];
 #endif
 #ifdef VLNEXT
-	mono_array_set (control_chars, gchar, 15, cc [VLNEXT]);
+	control_chars [15] = cc [VLNEXT];
 #endif
 #ifdef VEOL2
-	mono_array_set (control_chars, gchar, 16, cc [VEOL2]);
+	control_chars [16] = cc [VEOL2];
 #endif
 }
 
 MonoBoolean
-ves_icall_System_ConsoleDriver_TtySetup (MonoString *keypad, MonoString *teardown, MonoArray **control_chars, int **size)
+ves_icall_System_ConsoleDriver_TtySetup (MonoStringHandle keypad, MonoStringHandle teardown, MonoArrayHandleOut control_chars, int **size, MonoError* error)
 {
-	ERROR_DECL (error);
+	// FIXME Lock around the globals?
```

Comment:
Yea... `setup_finished` ought to use atomics or something. No need to fix in this PR.